frontend/src/components/Graph/yAxisConfig.ts

```diff
@@ -48,8 +43,33 @@ export const getYAxisFormattedValue = (
 
 export const getToolTipValue = (value: string, format: string): string => {
 	try {
+		let decimalPrecision: number | undefined;
+		const decimalSplitted = value.split('.');
+		if (decimalSplitted.length === 1) {
+			decimalPrecision = 0;
+		} else {
+			const decimalDigits = decimalSplitted[1].split('');
+			decimalPrecision = decimalDigits.length;
+			let nonZeroCtr = 0;
+			for (let idx = 0; idx < decimalDigits.length; idx += 1) {
+				if (decimalDigits[idx] !== '0') {
+					nonZeroCtr += 1;
+					if (nonZeroCtr >= 2) {
+						decimalPrecision = idx + 1;
+					}
+				} else if (nonZeroCtr) {
+					decimalPrecision = idx;
+					break;
+				}
+			}
+		}
 		return formattedValueToString(
-			getValueFormat(format)(parseFloat(value), undefined, undefined, undefined),
+			getValueFormat(format)(
```

Comment:
No need to pass `undefined` for optional parameters. It's implicitly understood. 

You can simply write, 
```
			getValueFormat(format)(
				parseFloat(value),
				decimalPrecision
			),
```